tensorflow/python/data/experimental/service/server_lib.py

```diff
@@ -227,6 +248,9 @@ def _stop(self):
   def __del__(self):
     self._stop()
 
+  def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
Let's follow the convention of https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/distribute/coordinator/cluster_coordinator.py.

We have `stop` and `join`, but `__exit__` is not needed.